src/rgw/rgw_realm_reloader.cc

```diff
@@ -98,6 +99,8 @@ void RGWRealmReloader::reload()
   rgw_log_usage_finalize();
 
   env.driver->shutdown();
+  // Let tasks finish before we destroy stuff
+  context_pool.finish();
```

Comment:
i really hope we can avoid having to stop/restart this context_pool. it's global to the application, so we could be running all kinds of stuff there that isn't scoped to the backend store/driver

the frontend may be the only other subsystem that's scheduling stuff there now, and `AsioFrontend::pause()` does stop any 'work' that would prevent `io_context::run()` from returning control. but what if we want to spawn other long-running coroutines in the background that aren't related to the backend? those shouldn't need to be interrupted by realm reload if they don't depend on the realm configuration

maybe you can say more about the lifetime issues you saw here that aren't resolved by the addition of shared_ptr in neorados?

Reply:
I wrote this part before I made the changes to neorados. I don't actually think the stop is needed any more. I'll take it out.